improver/nowcasting/optical_flow.py

```diff
@@ -206,6 +209,73 @@ def __repr__(self):
             self.point_weight,
         )
 
+    @staticmethod
+    def _check_input_cubes(cube1, cube2):
+        """Check that input cubes have appropriate and matching dimensions"""
+        # check the nature of the input cubes, and raise a warning if they are
+        # not both precipitation
+        if cube1.name() != cube2.name():
+            msg = "Input cubes contain different data types {} and {}"
+            raise ValueError(msg.format(cube1.name(), cube2.name()))
+
+        data_name = cube1.name().lower()
+        if "rain" not in data_name and "precipitation" not in data_name:
+            msg = (
+                "Input data are of non-precipitation type {}.  Plugin "
+                "parameters have not been tested and may not be appropriate"
+                " for this variable."
+            )
+            warnings.warn(msg.format(cube1.name()))
+
+        # check cubes have exactly two spatial dimension coordinates and a
+        # scalar time coordinate
+        check_input_coords(cube1, require_time=True)
+        check_input_coords(cube2, require_time=True)
+
+        # check cube dimensions match
+        if cube1.coord(axis="x") != cube2.coord(axis="x") or cube1.coord(
+            axis="y"
+        ) != cube2.coord(axis="y"):
+            raise InvalidCubeError("Input cubes on unmatched grids")
+
+        # check grids are equal area
+        check_if_grid_is_equal_area(cube1)
+        check_if_grid_is_equal_area(cube2)
+
+    @staticmethod
+    def _get_input_time_difference(cube1, cube2):
+        """Get time difference between input cubes, in seconds"""
+        time1 = (cube1.coord("time").units).num2date(cube1.coord("time").points[0])
+        time2 = (cube2.coord("time").units).num2date(cube2.coord("time").points[0])
```

Comment:
I know this is copied, but it can be written more neatly as:
```suggestion
        time1 = cube1.coord("time").cell(0).point
        time2 = cube2.coord("time").cell(0).point
```